document-store/src/main/java/org/hypertrace/core/documentstore/query/WhitelistedSelection.java

```diff
@@ -0,0 +1,42 @@
+package org.hypertrace.core.documentstore.query;
+
+import static org.hypertrace.core.documentstore.expression.Utils.validateAndReturn;
+
+import javax.validation.constraints.NotNull;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.hypertrace.core.documentstore.expression.impl.AggregateExpression;
+import org.hypertrace.core.documentstore.expression.type.SelectingExpression;
+
+/**
+ * A generic selection definition that supports expressions with optional aliases (used in the
+ * response).
+ */
+@Value
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class WhitelistedSelection implements Selection {
+
+  @NotNull SelectingExpression expression;
+
+  // Alias is optional. Handling missing aliases can be implemented in the respective parsers
+  String alias;
+
+  @Override
+  public boolean isAggregation() {
+    return expression instanceof AggregateExpression;
+  }
+
+  @Override
+  public boolean allColumnsSelected() {
+    return false;
+  }
+
+  public static Selection of(final SelectingExpression expression) {
+    return WhitelistedSelection.of(expression, null);
```

Comment:
why does validation not required in this method? Is validation is only for `alias`?

Reply:
Validation is required here. The invoked method `WhitelistedSelection.of()` does the validation internally.
Validation is only for the SelectingExpression. "alias" doesn't require any validation as it can be null/blank.